### ISSUE TITLE
synapse/config/__main__ uses wrong py2 import rules

### ISSUE THREAD

**Issue by hawkowl** [matrix-org/synapse#4350]
```
/opt/venvs/matrix-synapse/bin/python -m synapse.config
Traceback (most recent call last):
  File "/usr/lib/python3.5/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.5/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/opt/venvs/matrix-synapse/lib/python3.5/site-packages/synapse/config/__main__.py", line 19, in <module>
    from homeserver import HomeServerConfig
ImportError: No module named 'homeserver'
```

should be `from synapse.config.homeserver`

**Comment by hawkowl** [comment]
PR: https://github.com/matrix-org/synapse/pull/4356

**Comment by hawkowl** [comment]
Merged into develop, will be in 0.34.1.
